src/textCampaign/services/textCampaign.service.ts

```diff
@@ -0,0 +1,243 @@
+    const rumbleUpProjectData = {
+      name: createProjectDto.name,
+      msg: createProjectDto.message,
+      areacode: createProjectDto.areaCode,
+      group: createProjectDto.groupId,
+      flags: createProjectDto.flags,
+      outsource_start: createProjectDto.outsourceStart,
+      outsource_end: createProjectDto.outsourceEnd,
+      outsource_email: createProjectDto.outsourceEmail,
+    }
+
+    // Call RumbleUp API to create the project
+    const response =
+      await this.rumbleUpService.createProject(rumbleUpProjectData)
+
+    if (!response.success) {
+      throw new Error(
+        `Failed to create project in RumbleUp: ${response.error || response.message}`,
+      )
+    }
+
+    // Create a new TextCampaign record (no longer using upsert since we can have multiple per campaign)
+    const textCampaign = await this.prisma.textCampaign.create({
+      data: {
+        campaignId,
+        projectId: response.data?.id || null,
+        name: createProjectDto.name,
+        message: createProjectDto.message,
+      },
+    })
+
+    return textCampaign
+  }
+
+  /**
+   * Upload contacts from a CSV file and create a RumbleUp project
+   *
+   * This method performs two steps:
+   * 1. Uploads a CSV file of contacts to RumbleUp, which returns a group ID
+   * 2. Creates a project associated with the group ID
+   *
+   * The CSV file must have the following format:
+   * - The first line should be a header with column names
+   * - Required columns: "phone" and either "name" or "first_name"
+   * - Optional columns: last_name, email, street, city, zipcode, etc.
+   *
+   * @param campaignId The campaign ID to associate the project with
+   * @param csvData The CSV data as either:
+   *   - A file path string to a CSV file on disk
+   *   - A Buffer containing the CSV content
+   * @param projectDetails Project details (name, message, etc.) excluding groupId which will be generated
+   * @param fileName Optional filename if using Buffer (defaults to "contacts.csv")
+   * @returns The created TextCampaign record
+   */
+  async uploadContactsAndCreateProject(
+    campaignId: number,
+    csvData: string | Buffer,
+    projectDetails: Omit<CreateProjectSchema, 'groupId'>,
+    fileName?: string,
+  ) {
+    // Validate that the campaign exists
+    const campaign = await this.campaignsService.findFirst({
+      where: { id: campaignId },
+    })
+
+    if (!campaign) {
+      throw new NotFoundException(`Campaign with ID ${campaignId} not found`)
+    }
+
+    // Step 1: Upload the contacts CSV file and get the group ID
+    const groupId = await this.rumbleUpService.uploadContactsAndGetGroupId(
+      csvData,
+      fileName,
+    )
+
+    // Step 2: Create a project using the group ID
+    const createProjectDto: CreateProjectSchema = {
+      ...projectDetails,
+      groupId,
+    }
+
+    // Step 3: Create the project with the group ID
+    return this.createProject(campaignId, createProjectDto)
+  }
```

Comment:
Also.. when if and when we _do_ implement this...

This kind of orchestration between operations (upload, and then create) should be handled up in the controller layer, not down here in the service layer.

Have a `createContactGroup` method that takes the CSV, sends it to the import method in the `RumbleUpService` and returns the newly created `groupId`

Then, in the controller endpoint method, just call `createContactGroup`, and pass that `groupId` along to `createProject`